feature/gnoi/os/tests/osinstall/README.md

```diff
@@ -2,78 +2,96 @@
+           
+   3. Activate the software by issuing gnoi.os.Activate rpc as in the case of primary supervisor.
+      
+      a. Set the version field of the ActivateRequest message to be the same as the version specified in the TransferRequest message above.
+      
+      b. Set the no_reboot flag to true.
+      
+      c. Set the standby_supervisor to TRUE this time.
+      
+   4. Reboot the switch:
+      
+      a. Issue gnoi.system.Reboot as specified in [gNOI-3.1: Complete Chassis Reboot](feature/gnoi/tests/complete_chassis_reboot/complete_chassis_reboot_test.md).
+      
+   5. Verify that the supervisor image has moved to the new image:
+       
+      a. Verify that the supervisor has a valid image by issuing gnoi.os.Verify rpc.
+         * Expect a VerifyResponse with the version field set to the version specified in messages above eventually.
+         * Verify the standby supervisor version.
+         * Expect that the VerifyResponse.verify_standby has the same version in messages above.
+
+               
+* Subtest-2 : Configuration push verification post OS change and check health of the software-module that allows configuration.
+  1. Check the health of the software-module component that allows configuration of the router and verify if it is healthy using the leaf /components/component[**process that handles configuration of the DUT**]/healthz/status/
+     
+     a. If unhealthy, run HealthZ.Get() and HealthZ.Artifact() RPCs on the subject component to fetch artifacts corresponding to the event.
+        * Rollback to the previous OS version by following the steps in 3, 4 and 5 above from Sub-test-1 to recover the DUT from the faulty state.
+        * Mark the test as a failure due to issues with the OS upgrade and exit the test.
+          
+  2. Do a gNMI.GET() RPC to extract the current configuration on the DUT as backup.
+     
+  3. Push test configuration to the router using gNMI.Set() RPC with "replace operation" and reverify the status of the leaf /components/component[process that handles configuration of the DUT]/healthz/status/.
+     
+     a. If unhealthy, run HealthZ.Get() and HealthZ.Artifact() RPCs on the subject component to fetch artifacts corresponding to the event.
+        [TODO: Below step needs to be discussed to inderstand how to recover the DUT. May just need to depend on the Test infrastructure]
+        * Push the backup configuration fetched in Subtest-2 bullet#2 above to the DUT to recover the DUT.
+        * Mark the test as a failure due to issues with the OS upgrade and exit the test.
+          
+  4. Kill the process that manages device configuration using the "gNOI.KillProcessRequest_SIGNAL_ABRT" and restart flag set to False. This will terminate the process and will also dump a Core file, while maintaing the process in a down state. Verify if the leaf /components/component/healthz/state/ transitioned to unhealthy state.
```

Comment:
define the actual proto message you want here rather than just the "text"

Reply:
Updated to reflect the actual proto message.